pycbc/workflow/grb_utils.py

```diff
@@ -515,20 +515,24 @@ def make_pygrb_plot(workflow, exec_name, out_dir,
     if exec_name == 'pygrb_efficiency':
         # In this case tags[0] is the offtrial number
         node.add_input_list_opt('--seg-files', seg_files)
-        node.add_input_opt('--onsource-file',
-                           onsource_file)
         node.add_input_opt('--bank-file', bank_file)
-        node.new_output_file_opt(workflow.analysis_time, '.png',
-                                 '--background-output-file',
-                                 tags=extra_tags+['max_background'])
-        node.new_output_file_opt(workflow.analysis_time, '.png',
-                                 '--onsource-output-file',
-                                 tags=extra_tags+['onsource'])
-        node.new_output_file_opt(workflow.analysis_time, '.json',
-                                 '--exclusion-dist-output-file',
-                                 tags=extra_tags)
-        node.add_opt('--injection-set-name', tags[1])
         node.add_opt('--trial-name', tags[0])
+        node.add_opt('--injection-set-name', tags[1])
+        # Output the sensitivity plot
+        if kwargs['plot_bkgd']:
+            node.new_output_file_opt(workflow.analysis_time, '.png',
```

Comment:
Ah, okay, so `plot_bkgd` changes (completely I think) the output being produced. So this wouldn't be easily controlled in the config file (at least it wouldn't go in the [EXECUTABLE-TAG] section). I think my suggestion of subclassing PlotExectuable, and moving this code, and kwarg to the `create_node` method there would stand.

Reply:
Correct, it determines the output produced (exclusion distances vs detection efficiency), and I agree that `make_pygrb_plot` has grown too much as I adapted it to new plots to be shown.  I like the idea of separating it into subclasses of `PlotExectuable`.  I will open a PyGRB development issue, but will merge to keep this moving.